tests/components/romy/test_config_flow.py

```diff
@@ -0,0 +1,301 @@
+"""Test the ROMY config flow."""
+from ipaddress import ip_address
+from unittest.mock import AsyncMock, MagicMock, PropertyMock, patch
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components import zeroconf
+from homeassistant.components.romy.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+
+def _create_mocked_romy(
+    is_initialized,
+    is_unlocked,
+    user_name="MyROMY",
+    unique_id="aicu-aicgsbksisfapcjqmqjq",
+    model="005:000:000:000:005",
+    port=8080,
+):
+    mocked_romy = AsyncMock(MagicMock)
+    type(mocked_romy).is_initialized = PropertyMock(return_value=is_initialized)
+    type(mocked_romy).is_unlocked = PropertyMock(return_value=is_unlocked)
+    type(mocked_romy).user_name = PropertyMock(return_value=user_name)
+    type(mocked_romy).unique_id = PropertyMock(return_value=unique_id)
+    type(mocked_romy).port = PropertyMock(return_value=port)
+    type(mocked_romy).model = PropertyMock(return_value=model)
+
+    # Mock async methods
+    async def mock_set_name(new_name):
+        mocked_romy.name = PropertyMock(return_value=new_name)
+        return True, '{"success": true}'
+
+    type(mocked_romy).set_name = AsyncMock(side_effect=mock_set_name)
+
+    return mocked_romy
+
+
+CONFIG = {CONF_HOST: "1.2.3.4", CONF_PASSWORD: "12345678"}
+
+INPUT_CONFIG_HOST = {
+    CONF_HOST: CONFIG[CONF_HOST],
+}
+
+INPUT_CONFIG_PASS = {
+    CONF_PASSWORD: CONFIG[CONF_PASSWORD],
+}
+
+
+# user conf tests
+###################
+async def test_show_user_form(hass: HomeAssistant) -> None:
+    """Test that the user set up form with config."""
+
+    mocked_romy = _create_mocked_romy(
+        is_initialized=True,
+        is_unlocked=True,
+    )
+
+    with patch(
+        "homeassistant.components.romy.config_flow.romy.create_romy",
+        return_value=mocked_romy,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data=INPUT_CONFIG_HOST,
+        )
+
+    assert "errors" not in result
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+
+
+async def test_show_user_form_with_wrong_host(
+    hass: HomeAssistant,
+) -> None:
+    """Test that the user enters wrong host."""
+
+    mocked_romy = _create_mocked_romy(
+        is_initialized=False,
+        is_unlocked=False,
+    )
+
+    with patch(
+        "homeassistant.components.romy.config_flow.romy.create_romy",
+        return_value=mocked_romy,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data=INPUT_CONFIG_HOST,
+        )
+
+    assert result["errors"].get("host") == "cannot_connect"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
```

Comment:
Config flow test should always be finished by creating an entry or with an abort.

Reply:
rewrote tests, all are ending now with create or abort entry